modules/basics/src/main/java/com/opengamma/strata/basics/schedule/StubConvention.java

```diff
@@ -259,7 +257,7 @@ public RollConvention toRollConvention(
     // dates are at the end of the month, and in different months
     if (this == NONE) {
       if (start.getDayOfMonth() != end.getDayOfMonth() &&
-          start.getLong(PROLEPTIC_MONTH) != end.getLong(PROLEPTIC_MONTH) &&
+          start.getMonth() != end.getMonth() &&
```

Comment:
This seems like an even odder test than before. A schedule from July to July will handled differently to one from July to August.

Maybe keep the existing code, but inside the block check `preferEndOfMonth` and return `EOM` if true?

Reply:
Per the existing comment it looks like the original intention here was to handle the case where the months are different, so varying behaviour when months are different vs the same doesn't seem unexpected?

An effect of the above suggestion would be that an EOM trade with start on 28 February and end 30th April would change to return EOM roll convention rather than 30. This sounds fine in the context of "preferEndOfMonth" being true but does feel like a slightly more intrusive change in behaviour than currently proposed.